src/Stratis.Bitcoin.Cli/Program.cs

```diff
@@ -10,53 +14,136 @@ namespace Stratis.Bitcoin.Cli
+                        RPCResponse response = rpc.SendCommand(args.ElementAt(1), args.Skip(2).ToArray());
 
-                // Return the result as a string to the console
-                Console.WriteLine(response.ResultString);
+                        // Return the result as a string to the console.
+                        Console.WriteLine(response.ResultString);
+                    }
+                    catch (Exception err)
+                    {
+                        Console.WriteLine(err.Message);
+                    }
+                }
+                else
+                {
+                    // Process API call.
+                    using (var client = new HttpClient())
+                    {
+                        client.DefaultRequestHeaders.Accept.Clear();
+                        client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                        var url = $"http://localhost:{apiPort}/api/{args.ElementAt(1)}?{string.Join("&", args.Skip(2))}";
+                        try
+                        {
+                            // Get the response.
+                            Console.WriteLine($"Sending API command to {url}...");
+                            var response = client.GetStringAsync(url).GetAwaiter().GetResult();
+                            
+                            // Format and return the result as a string to the console.
+                            Console.WriteLine(JsonConvert.SerializeObject(JsonConvert.DeserializeObject<object>(response), Formatting.Indented));
+                        }
+                        catch (Exception err)
+                        {
+                            Console.WriteLine(ExceptionToString(err));
+                        }
+                    }
+                }
             }
             catch (Exception err)
             {
-                // Report any errors to the console
-                Console.WriteLine(err.ToString());
+                // Report any errors to the console.
+                Console.WriteLine(ExceptionToString(err));
+            }
+        }
+
+        /// <summary>
+        /// Determine both the exception message and any inner exception messages.
+        /// </summary>
+        /// <param name="exception">The exception object.</param>
+        /// <returns>Returns the exception message plus any inner exceptions.</returns>
+        public static string ExceptionToString(Exception exception)
+        {
+            bool isDebugMode = false;
+#if DEBUG
+            isDebugMode = true;
+#endif
+            Exception ex = exception;
+            StringBuilder stringBuilder = new StringBuilder(128);
+            while (ex != null)
+            {
+                if (isDebugMode)
+                {
+                    stringBuilder.Append(ex.GetType().Name);
```

Comment:
you can probably merge these 2 lines in one statement.

Reply:
Done.